lib/kubeclient_next/client/rest_client.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require "faraday"
+require "base64"
+
+module KubeclientNext
+  module Client
+    class RESTClient
+      attr_reader :group, :version
+
+      def initialize(config:, context:, group:, version:)
+        @config = config
+        @context = context
+        @group = group
+        @version = version
+
+        cert_store = OpenSSL::X509::Store.new
+        cert_store.add_cert(OpenSSL::X509::Certificate.new(Base64.decode64(cluster.certificate_authority_data)))
+        @connection = Faraday.new(
+          endpoint,
+          ssl: {
+            client_cert: OpenSSL::X509::Certificate.new(Base64.decode64(user.auth_info.client_certificate_data)),
+            client_key: OpenSSL::PKey::RSA.new(Base64.decode64(user.auth_info.client_key_data)),
+            cert_store: cert_store,
+          }
+        )
+      end
+
+      def get_namespaces
+        @connection.get("namespaces")
+      end
+
+      private
+
+      attr_reader :config, :context, :connection
+
+      # Core/v1 resources are in `/api/#{version}`. In general, all other resources are found in `/apis/GROUP/VERSION`
+      def endpoint
+        @endpoint ||= group ? URI.join(host, @group, @version) : URI.join(host, "/api/#{version}")
+      end
```

Comment:
It may be worthwhile to expose an even lower-level REST client, one that can query arbitrary endpoints of the API server.